core/object/script_language.cpp

```diff
@@ -154,21 +156,45 @@ void ScriptServer::unregister_language(ScriptLanguage *p_language) {
 	}
 }
 
-void ScriptServer::init_languages() {
-	{ //load global classes
-		global_classes_clear();
-		if (ProjectSettings::get_singleton()->has_setting("_global_script_classes")) {
-			Array script_classes = ProjectSettings::get_singleton()->get("_global_script_classes");
-
-			for (int i = 0; i < script_classes.size(); i++) {
-				Dictionary c = script_classes[i];
-				if (!c.has("class") || !c.has("language") || !c.has("path") || !c.has("base")) {
-					continue;
-				}
-				add_global_class(c["class"], c["base"], c["language"], c["path"]);
+String ScriptServer::_get_global_script_class(const String &p_type, const String &p_path, StringName *r_lang, String *r_extends, String *r_icon_path) {
+	for (int i = 0; i < ScriptServer::get_language_count(); i++) {
+		if (ScriptServer::get_language(i)->handles_global_class_type(p_type)) {
+			String global_name = ScriptServer::get_language(i)->get_global_class_name(p_path, r_extends, r_icon_path);
+			*r_lang = ScriptServer::get_language(i)->get_name();
+			return global_name;
+		}
+	}
+	*r_lang = String();
+	*r_extends = String();
+	*r_icon_path = String();
+	return String();
+}
+
+void ScriptServer::_scan_script_classes(const String &p_path) {
+	_Directory dir;
+	dir.open(p_path);
+	dir.list_dir_begin(true);
+	String file;
+	while (!(file = dir.get_next()).empty()) {
+		if(dir.current_is_dir()) {
+			// TODO: Hard coded path delimiter
+			_scan_script_classes(p_path + "/" + file);
```

Comment:
Put this TODO because

a) The string concatenation seemed unusual to me
b) Considering whether or not `_scan_script_classes` should take a property not of type `String`

Reply:
For future reference, this should have been `p_path.plus_file(file)`